### PR TITLE
Fix install global

### PR DESCRIPTION
**Pull request by victorhurdugaci**
Fixes https://github.com/aspnet/DNX/issues/1287
Fixes https://github.com/aspnet/DNX/issues/1267

- Change global install to use restore when installing
- Uninstall uses lock file to decide what is still used

Please review: @davidfowl @ChengTian @lodejard 

@tugberkugurlu want to try this before I merge it to see if your scenario works now?

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@victorhurdugaci__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Victor Hurdugaci). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla2.msopentech.com.
    </p>

TTYL, MSOTBOT;

File: src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs [hunk 1]
```diff
@@ -40,42 +42,179 @@ public FeedOptions FeedOptions
...
+                }
+            }
+            finally
+            {
+                var folderToDelete = Path.GetDirectoryName(temporaryProjectFileFullPath);
+                FileOperationUtils.DeleteFolder(folderToDelete);
+                Directory.Delete(folderToDelete);
+            }
+
+            var packageFullPath = Path.Combine(
```
Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: Use `DefaultPackagePathResolver` to get install path

File: src/Microsoft.Framework.PackageManager/Install/UninstallCommand.cs [hunk 2]
```diff
@@ -64,43 +67,52 @@ public UninstallCommand(IApplicationEnvironment applicationEnvironment, IAppComm
...
             {
-                FrameworkName = _environment.RuntimeFramework,
-                ProjectLibraryProviders = new List<IWalkProvider>(),
-                RemoteLibraryProviders = new List<IWalkProvider>(),
-                LocalLibraryProviders = new IWalkProvider[] {
-                    new LocalWalkProvider(
-                        new NuGetDependencyResolver(
-                            new PackageRepository(
-                                packagesRepo.RepositoryRoot.Root)))
+                var appLockFileFullPath = Path.Combine(
```
Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Install/UninstallCommand.cs]: Use `DefaultPackagePathResolver` to get package install path

File: src/Microsoft.Framework.PackageManager/Install/UninstallCommand.cs [hunk 3]
```diff
@@ -64,43 +67,52 @@ public UninstallCommand(IApplicationEnvironment applicationEnvironment, IAppComm
...
+                    _commandsRepo.PackagesRoot.Root,
+                    applicationPackage.Id,
+                    applicationPackage.Version.ToString(),
+                    InstallBuilder.CommandsFolderName,
+                    LockFileFormat.LockFileName);
+
+                if (!File.Exists(appLockFileFullPath))
+                {
+                    // The lock file is missing, probably the app is not installed correctly
```
Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Install/UninstallCommand.cs]: Do we want to show warning here?

**Comment by tugberkugurlu** [comment]
Nice, I will definitely give it a go!

Date: Tue, 10 Mar 2015 17:08:43 -0700

**Comment by tugberkugurlu** [comment]
I was not able to build the **FixGlobalInstall** branch. Getting this (obviously this is screenshot is not gonna be helpful):

image

Are you guys interested in getting the entire build log?

**Comment by victorhurdugaci** [comment]
There were changes between the moment I pushed and now. Logging had some api renames. I'll rebase tomorrow morning and update with the changes in logging

Sent from my phone
________________________________

**Comment by victorhurdugaci** [comment]
@tugberkugurlu I rebased and updated to the latest code changes. Want to try now?

**Comment by tugberkugurlu** [comment]
@victorhurdugaci nice, will try now!

**Comment by victorhurdugaci** [comment]
If you run the tests, there will be 4 Bundle tests that fail. It is a known issue and it is not related.

**Comment by tugberkugurlu** [comment]
I am still getting build errors :disappointed: 

image

I believe I am on the right commit:

image

I am on 1.0.0-beta4-11257, clr, x86.

**Comment by victorhurdugaci** [comment]
did you run `git submodule init` and `git submodule update` ?

**Comment by tugberkugurlu** [comment]
oh, dear! I didn't even know that :smile: Should have read the readme first :smile: build is working now.

File: src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs [hunk 4]
```diff
@@ -40,67 +42,212 @@ public FeedOptions FeedOptions
...
             }
 
+            return true;
+        }
+
+        private bool InstallCommands(string packageId, string appPath)
+        {
+            string commandsFolder = Path.Combine(appPath, InstallBuilder.CommandsFolderName);
+
+            var fileFilter = PlatformHelper.IsWindows ? "*.cmd" : "*.sh";
```
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: BTW, we don't call the file .sh anywhere else. We should remove that. kpm bundle today makes foo.cmd and foo.

File: src/Microsoft.Framework.PackageManager/Install/UninstallCommand.cs [hunk 5]
```diff
@@ -6,6 +6,9 @@
 using System.Threading.Tasks;
 using Microsoft.Framework.Runtime;
 using NuGet;
+using System;
```
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Install/UninstallCommand.cs]: Usings are unsorted.

**Comment by tugberkugurlu** [comment]
OK, new problem :smile: ran `build dev-install`, I am using the right thing:

> Active Version           Runtime Architecture Location
> ------ -------           ------- ------------ --------
>        1.0.0-beta4-11249 clr     x86          C:\Users\tugberk.ugurlu\.dnx\r...
>        1.0.0-beta4-11249 coreclr x86          C:\Users\tugberk.ugurlu\.dnx\r...
>        1.0.0-beta4-11257 clr     x86          C:\Users\tugberk.ugurlu\.dnx\r...
>   *    1.0.0-dev         clr     x86          C:\Users\tugberk.ugurlu\.dnx\r...
>        1.0.0-dev         coreclr x86          C:\Users\tugberk.ugurlu\.dnx\r...
>        1.0.0-dev         mono                 C:\Users\tugberk.ugurlu\.dnx\r...

but I don't have kpm, dnx under my path as the build artifacts doesn't have them. Tried to copy kpm from the other runtime versions but it still need dnx.exe.

**Comment by victorhurdugaci** [comment]
Did the build succeed?

**Comment by tugberkugurlu** [comment]
I believe so:

image

"C:\Users\tugberk.ugurlu\.dnx\runtimes\dnx-clr-win-x86.1.0.0-dev\bin" doesn't contain kpm and dnx.exe.

**Comment by tugberkugurlu** [comment]
@victorhurdugaci Jabbr?

**Comment by victorhurdugaci** [comment]
@tugberkugurlu I'm there (AspNetvNext)

**Comment by tugberkugurlu** [comment]
I was able to get it building but I cannot get the project building now (had to rename xunit.runner.kre to .aspnet). Here is the version I am trying: https://github.com/tugberkugurlu/k-http/blob/92681b7bbf49c4c55a74f1b418b489cede1dddc1/src/k-http/program.cs

Getting this:

image

which is weird. Any clue? it says I am on dnx 1.0.0-t150313204748 (which is a dev build).

**Comment by victorhurdugaci** [comment]
Update project.json with the new moniker for `aspnet50`

**Comment by tugberkugurlu** [comment]
All looks fine. I don't see the issue I was seeing on #1287 and  #1267.

Am I allowed to use :shipit:? :trollface:

File: src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs (hunk 4 above)
Comment (victorhurdugaci) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: Ah, I remember why I added the `.sh` extension. The NuGet package builder doesn't support file without extensions. I'm not sure if it is a bug or not 

```
System.ArgumentOutOfRangeException: startIndex cannot be larger than length of string.
Parameter name: startIndex
   at System.String.Substring(Int32 startIndex, Int32 length)
   at System.String.Substring(Int32 startIndex)
   at NuGet.PackageBuilder.WriteFiles(ZipArchive package) in D:\aspnet\dnx\src\Microsoft.Framework.Runtime\NuGet\Authori
ng\PackageBuilder.cs:line 468
   at NuGet.PackageBuilder.Save(Stream stream) in D:\aspnet\dnx\src\Microsoft.Framework.Runtime\NuGet\Authoring\PackageB
uilder.cs:line 257
   at Microsoft.Framework.PackageManager.BuildManager.Build() in D:\aspnet\dnx\src\Microsoft.Framework.PackageManager\Bu
ilding\BuildManager.cs:line 184
   at Microsoft.Framework.PackageManager.Program.<>c__DisplayClass3_3.<Main>b__6() in D:\aspnet\dnx\src\Microsoft.Framew
ork.PackageManager\Program.cs:line 162
   at Microsoft.Framework.Runtime.Common.CommandLine.CommandLineApplication.Execute(String[] args) in D:\aspnet\dnx\src\
Microsoft.Framework.CommandLineUtils\CommandLine\CommandLineApplication.cs:line 277
   at Microsoft.Framework.PackageManager.Program.Main(String[] args) in D:\aspnet\dnx\src\Microsoft.Framework.PackageMan
```

**Pull request merged**